### PR TITLE
Setting up tests and documentation

### LINKED ISSUE #1: Add repo documentation, tests
@eccuraa

### PR DESCRIPTION
**Pull request by juaristi22**
Fix #1

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,14 +1,18 @@
 [project]
 name = "microcalibrate"
 version = "0.1.0"
-description = "Add your description here"
+description = "Calibrating microdata"
 readme = "README.md"
 authors = [
     { name = "Nikhil Woodruff", email = "nikhil.woodruff@outlook.com" }
```
Comment (nikhilwoodruff) [inline pyproject.toml:7]: Add yourself!

File: src/microcalibrate/config.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+# Define a configuration for pydantic validation that allows
+# arbitrary types like pd.DataFrame
+VALIDATE_CONFIG = ConfigDict(arbitrary_types_allowed=True)
+
+DEFAULT_LOG_FORMAT = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+DEFAULT_DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+# Define logging levels for different environments
+DEBUG_LEVEL = logging.DEBUG
```
Comment (nikhilwoodruff) [inline src/microcalibrate/config.py:22]: Do we need this? Why can't we just use `logging.warning` etc.

File: src/microcalibrate/reweight.py [hunk 3]
```diff
@@ -11,16 +17,41 @@
...
-):
+    original_weights: np.ndarray,
+    loss_matrix: pd.DataFrame,
+    targets_array: np.ndarray,
+    dropout_rate: Optional[float] = 0.1,
+    epochs: Optional[int] = 2_000,
+    noise_level: Optional[float] = 10.0,
+    subsample_every: Optional[int] = 50,
+    learning_rate: Optional[float] = 1e-3,
+) -> tuple[np.ndarray, np.ndarray]:
```
Comment (nikhilwoodruff) [inline src/microcalibrate/reweight.py:28]: Can we add a log level default to error: by default I think we should produce only the progress bar

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# MicroCalibrate Tests
```
Comment (nikhilwoodruff) [inline tests/README.md:1]: Capitalisation please

File: tests/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- **test_basic.py**: Basic functionality tests for the package
+- Coming soon...
+
+## Running Tests
+
+To run the tests, use the following command from the project root:
+
+```bash
+python -m pytest tests/
```
Comment (nikhilwoodruff) [inline tests/README.md:17]: Can we make all common CLI commands `make test` etc.

File: tests/test_basic.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+"""Basic tests to verify CI setup."""
+
+
+def test_import() -> None:
+    """Test that the package can be imported."""
+    import microcalibrate
+
+    assert microcalibrate is not None
```
Comment (nikhilwoodruff) [inline tests/test_basic.py:8]: Think this is unneeded- it we can't import it, we'll get the error on the previous line.

File: tests/test_calibration.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        epochs=528,
+        learning_rate=0.01,
+        dropout_rate=0,
+        subsample_every=0,
+    )
+
+    # Call calibrate method on our data and targets of interest
+    calibrator.calibrate()
+
+    final_weights = targets_matrix.mul(calibrator.weights, axis=0).sum().values
```
Comment (nikhilwoodruff) [inline tests/test_calibration.py:55]: Rename to something else please, like `final_estimates` since this is inaccurate

**Review by nikhilwoodruff — CHANGES_REQUESTED** [review]
Thanks! Questions and changes requested

**Review by nikhilwoodruff — APPROVED** [review]
